extensions/azurecore/src/azureResource/commands.ts

```diff
@@ -181,17 +181,22 @@ export function registerAzureResourceCommands(appContext: AppContext, azureViewT
 		vscode.commands.executeCommand('workbench.actions.modal.linkedAccount');
 	});
 
-	vscode.commands.registerCommand('azure.resource.connectsqlserver', async (node?: TreeNode) => {
+	vscode.commands.registerCommand('azure.resource.connectsqlserver', async (node?: TreeNode | azdata.ObjectExplorerContext) => {
 		if (!node) {
 			return;
 		}
-
-		const treeItem: azdata.TreeItem = await node.getTreeItem();
-		if (!treeItem.payload) {
-			return;
+		let connectionProfile: azdata.IConnectionProfile = undefined;
+		if (node instanceof TreeNode) {
+			const treeItem: azdata.TreeItem = await node.getTreeItem();
+			if (!treeItem.payload) {
+				return;
+			}
+			// Ensure connection is saved to the Connections list, then open connection dialog
+			connectionProfile = Object.assign({}, treeItem.payload, { saveProfile: true });
```

Comment:
I still think we want to keep this logic for the other case as well, since the idea is we want to open up the connection dialog and save the connection to OE when we do this action.

Reply:
the connection profile already have the saveProfile set, I checked

Reply:
Are you sure we can guarantee that and it's didn't just happen to have it that time? There's no harm in setting it again regardless so I'd say we should still do that.

Reply:
Actually I just noticed that we pass in saveConnection : true to the call to openConnectionDialog below...so now I'm left wondering why this one is even necessary. But I still think we should be consistent at least so either remove it from both or add it to both.